src/devices/Common/Iot/Device/Graphics/BdfFont.cs

```diff
@@ -0,0 +1,266 @@
+        // private static readonly string s_sWidth             = "SWIDTH";
+        // private static readonly string s_dWidth             = "DWIDTH";
+        private static readonly string s_bbx                = "BBX ";
+        // private static readonly string s_vVector            = "VVECTOR";
+        private static readonly string s_endChar            = "ENDCHAR";
+        private static readonly string s_bitmap             = "BITMAP";
+
+        private BdfFont() { }
+
+        public static BdfFont Load(string fontFilePath)
+        {
+            using (StreamReader sr = new StreamReader(fontFilePath))
+            {
+                BdfFont font = new BdfFont();
+                while (!sr.EndOfStream)
+                {
+                    ReadOnlySpan<char> span = sr.ReadLine().AsSpan().Trim();
+                    if (span.StartsWith(s_fontBoundingBox.AsSpan(), StringComparison.Ordinal))
+                    {
+                        span = span.Slice(s_fontBoundingBox.Length).Trim();
+                        font.Width = ReadNextDecimalNumber(ref span);
+                        font.Height = ReadNextDecimalNumber(ref span);
+                        font.XDisplacement = ReadNextDecimalNumber(ref span);
+                        font.YDisplacement = ReadNextDecimalNumber(ref span);
+                        font.BytesPerGlyph = (int) Math.Ceiling(((double)font.Width) / 8);
+                    }
+                    else if (span.StartsWith(s_charSet.AsSpan(), StringComparison.Ordinal))
+                    {
+                        span = span.Slice(s_charSet.Length).Trim();
+                        if (span.CompareTo(s_isoCharset.AsSpan(), StringComparison.Ordinal) != 0)
+                        {
+                            throw new NotSupportedException("We only support ISO10646 for now.");
+                        }
+                    }
+                    else if (span.StartsWith(s_defaultChar.AsSpan(), StringComparison.Ordinal))
+                    {
+                        span = span.Slice(s_defaultChar.Length).Trim();
+                        font.DefaultChar = ReadNextDecimalNumber(ref span);
+                    }
+                    else if (span.StartsWith(s_Chars.AsSpan(), StringComparison.Ordinal))
+                    {
+                        span = span.Slice(s_Chars.Length).Trim();
+                        font.CharsCount = ReadNextDecimalNumber(ref span);
+
+                        if (font.Width == 0 || font.Height == 0 || font.CharsCount <= 0)
+                        {
+                            throw new InvalidDataException("The font data is not well formed.");
+                        }
+
+                        font.ReadGlyphsData(sr);
+                    }
+                }
+
+                return font;
+            }
+        }
+
+        public void GetCharData(char c, out ReadOnlySpan<ushort> charData)
+        {
+            if (!GlyphMapper.TryGetValue((int) c, out int index))
```

Comment:
nit: extra space between `(int)` and `c` (same line below)

Reply:
personally I prefer to have space. But, I'll change it just to be consistent.